test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ModelStateDictionaryExtensionsTest.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class ModelStateDictionaryExtensionsTest
+    {
+        [Fact]
+        public void AddsModelErrorMessage_ForSingleExpression()
```

Comment:
We try to include expectations in test names. So here the name should be `AddModelError_ForSingleExpression_AddsExpectedMessage`.